src/app/mesh/qgsmeshtimeformatdialog.cpp

```diff
@@ -96,4 +104,22 @@ void QgsMeshTimeFormatDialog::enableGroups( bool useAbsoluteTime )
   mRelativeTimeGroupBox->setEnabled( ! useAbsoluteTime );
 }
 
+bool QgsMeshTimeFormatDialog::layerHasReferenceTime() const
+{
+  if ( !mLayer )
+    return false;
+  int groupCount = mLayer->dataProvider()->datasetGroupCount();
+  bool found = false;
```

Comment:
you can remove this variable and simplify this to 

``` 
for ( int i=0; i< mLayer->dataProvider()->datasetGroupCount(); ++i)
  if (mLayer->dataProvider()->datasetGroupMetadata( i ).referenceTime().isValid())
    return true;

return false;
```

Reply:
thinking about it, maybe you can return here the index of dataset group that HAS the reference time, and -1 if not found. that way you do not need to search second time for this group